rewrite-java/src/main/java/org/openrewrite/java/JavaPrinter.java

```diff
@@ -831,6 +831,13 @@ public J visitMultiCatch(MultiCatch multiCatch, P p) {
         return multiCatch;
     }
 
+    @Override
+    public J visitVarType(VarType varType, P p) {
+        visitMarkers(varType.getMarkers(), p);
+        getPrinter().append("var");
+        return varType;
+    }
```

Comment:
Something feels not quite right about this...
Whatever part of JavaPrinter that is printing the whitespace - which some part of JavaPrinter clearly is since your tests pass - from the `VarType` is probably also visiting its markers. If that hunch is correct, any search results would end up printed twice. Can you double check that isn't the case? Maybe slap a search marker onto a `VarType` as part of your test, just to be sure.

Reply:
yeah, good feedback, let me do that.